### ISSUE TITLE
GRPO reference model reports nonzero KL on the first step, when policy and reference are identical

### ISSUE THREAD

**Issue by haanjack** [haanjack/ironcore#96]
`_create_reference_model` (non-FSDP path, `ironcore/trainers/grpo_trainer.py:206-215`) is `copy.deepcopy(model)` — the reference starts out bit-identical to the policy. On the first step, before any update, `kl_loss` should therefore be exactly 0.

It is not. Running the GRPO smoke config against Qwen2.5-0.5B (as it was before #88 shrank it):

```
step: 1, grpo_loss: 1.3144, policy_loss: 0.0000, kl_loss: 1.3144, mean_reward: 0.0000, mean_ratio: 1.0000
step: 2, grpo_loss: 2.1844, policy_loss: 0.0000, kl_loss: 2.1844, mean_reward: 0.0000, mean_ratio: 1.0000
step: 3, grpo_loss: 1.4679, policy_loss: 0.0000, kl_loss: 1.4679, ...
step: 4, grpo_loss: 3.9359, policy_loss: 0.0000, kl_loss: 3.9359, ...
```

The same config at 19.9M parameters gives `kl_loss: 0.0000` throughout, which is the value the deep-copy implies.

Leading hypothesis, not yet confirmed: the reference is put in `eval()` while the policy runs in `train()` with `dropout_attn`/`dropout_mlp`/`dropout_embd` all defaulting to 0.1. The two then evaluate the same weights under different dropout, and the difference is reported as divergence from the reference. If so the KL penalty is measuring dropout noise rather than policy drift, and `grpo_beta` is regularising against nothing.

Worth checking as well: `mean_ratio` stays exactly 1.0000 and `policy_loss` exactly 0.0000 across every step of this config, so with `grpo_num_epochs: 1` the smoke test never exercises the importance-ratio path at all. Setting `grpo_num_epochs: 2` moves `mean_ratio` to 0.0005 — a three-orders-of-magnitude swing after a single optimizer step at `max_lr: 1e-5`, which does not look right either and deserves its own look.

Found while measuring why that smoke test needed 7 GiB (#88). Filing rather than fixing because the right behaviour here is a modelling decision: whether the KL reference should see dropout at all.

**Comment by haanjack** [comment]
Confirmed: it is dropout. Same 0.5B config, only the three `dropout_*` rates changed.

```
dropout 0.1   step 1  kl_loss: 1.3144   step 2  2.1844   step 3  1.4679   step 4  3.9359
dropout 0.0   step 1  kl_loss: 0.0000   step 2  0.0001   step 3  0.0025   step 4  0.0001
```

`_create_reference_model` deep-copies the policy and calls `.eval()`, so the reference never applies dropout. The policy's log-prob pass during the update runs in `train()`, so it does. The two therefore disagree on identical weights, and the whole of that disagreement is charged as divergence from the reference.

So `grpo_beta` is regularising against dropout noise rather than policy drift. In this config `policy_loss` is 0 (the keyword reward never fires, so advantages are flat), which means the KL term *is* the entire loss — the run is optimising against its own dropout masks.

`ModelConfig` defaults `dropout_attn`, `dropout_mlp` and `dropout_embd` to 0.1, and none of the shipped GRPO configs override them, so this is the default path rather than a corner case.

Two more notes:

- The 19.9M model reports `kl_loss: 0.0000` with dropout still at 0.1. That is not the bug being absent — at random init the logits are near-uniform, so dropout barely moves the distribution. The effect grows with how peaked the logits are, which means it is **worst on a real pretrained policy**, exactly where it matters.
- Disabling dropout during RL fine-tuning is the usual remedy in this situation, so the fix is probably to force it off for GRPO (loudly, not silently) rather than to change how the reference model is built. That is a modelling call, hence still filing rather than fixing.
